eclair-core/src/main/scala/fr/acinq/eclair/Eclair.scala

```diff
@@ -393,4 +404,24 @@ class EclairImpl(appKit: Kit) extends Eclair {
     val sendPayment = SendPaymentRequest(amount, paymentHash, recipientNodeId, maxAttempts, externalId = externalId_opt, routeParams = Some(routeParams), userCustomTlvs = keySendTlvRecords)
     (appKit.paymentInitiator ? sendPayment).mapTo[UUID]
   }
+
+  override def signMessage(message: String): SignedMessage = {
+    val bytesToSign = SignedMessage.signedBytes(message)
+    val (signature, recoveryId) = appKit.nodeParams.keyManager.signDigest(bytesToSign)
+    SignedMessage(appKit.nodeParams.nodeId, message, (recoveryId + 31).toByte +: signature)
```

Comment:
Shouldn't this be:

```suggestion
    SignedMessage(appKit.nodeParams.nodeId, message, signature :+ (recoveryId + 31).toByte)
```

Reply:
In invoices we put the recoveryId at the end of the signature, what does lnd and c-lightning do for this API?

Reply:
They put it at the front. See https://twitter.com/rusty_twit/status/1182102005914800128 #specinatweet